src/index.js

```diff
@@ -28,15 +29,32 @@ module.exports = {
     start: compileProject,
     watcher: async function ({ filename, /* event, */ inventory }) {
       if (filename.endsWith('.ts') || filename.endsWith('.tsx')) {
-        // Second pass filter by Lambda dir
+        let { cwd } = inventory.inv._project
+        let globalTsConfig = getTsConfig(cwd)
+        let tsCompilerOptions = require(globalTsConfig).compilerOptions
+        if (tsCompilerOptions) {
+          let recompileProject = false
+          let tsPaths = tsCompilerOptions.paths || {}
+          for (const [_, paths] of Object.entries(tsPaths)) {
+            paths.map((p) => {
+              if (filename.startsWith(resolve(cwd, tsCompilerOptions.baseUrl, p).replace(/\/?\*$/, ""))) {
+                recompileProject = true
+              }
+            })
+          }
+
+          if (recompileProject) {
+            compileProject({ inventory })
+            return
+          }
+        }
+
```

Comment:
This looks good otherwise, although we still need tests!

Reply:
@ryanblock I've updated the logic to:

- take care of windows path
- take care of non TS/TSX paths alias